server/server.go

```diff
@@ -27,38 +27,39 @@ func New(db db.SQLClient, authority auth.Authenticator, cf cloudfoundry.Client)
 	// Never crash on panic
 	e.Use(middleware.Recover())
 
-	// Require a token for all requests
-	e.Use(auth.UAATokenAuthentication(authority))
-
 	// Validate and parse range query param
 	e.Use(api.ValidateRangeParams)
 
+	authGroup := e.Group("", auth.UAATokenAuthentication(authority))
```

Comment:
These `Group` are much prettier 👍

Reply:
I changed that to put the routes directly in the echo middleware.

Reply:
Looking at the code I think that `UAATokenAuthentication` middleware needs to match on `c.Request().URL.Path` not `c.Path()`

Reply:
BTW, I think we do not have tests for this? shall we add tests?

Reply:
Would be nice to avoid these regressions, however the UI and browser oauth-flow _should_ be temporary in paas-billing ... the end game is that the UI to view usage/billing would be in paas-admin and paas-billing will just be a JSON api and won't need to care about login and we can drop this code.